tests/test_scalars.py

```diff
@@ -156,116 +181,131 @@ def test_timedelta_add_sub() -> None:
         assert_type(
-            td + as13,  # pyright: ignore[reportGeneralTypeIssues]
+            td + as_ndarray_dt64,  # pyright: ignore[reportGeneralTypeIssues]
             npt.NDArray[np.datetime64],
         ),
         np.ndarray,
+        np.datetime64,
     )
-    check(assert_type(td + as14, NaTType), NaTType)
-
-    check(assert_type(as1 + td, pd.Period), pd.Period)
-    check(assert_type(as2 + td, pd.Timestamp), pd.Timestamp)
-    check(assert_type(as3 + td, dt.datetime), dt.datetime)
-    check(assert_type(as4 + td, dt.date), dt.date)
-    check(assert_type(as5 + td, pd.Timestamp), pd.Timestamp)
-    # pyright is wrong here because as6 + td calls td.__radd__(as6),
-    # not timedelta.__add__
+    check(assert_type(td + as_nat, NaTType), NaTType)
+
+    check(assert_type(as_period + td, pd.Period), pd.Period)
+    check(assert_type(as_timestamp + td, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_datetime + td, dt.datetime), dt.datetime)
+    check(assert_type(as_date + td, dt.date), dt.date)
+    check(assert_type(as_datetime64 + td, pd.Timestamp), pd.Timestamp)
+    # pyright can't know that as6 + td calls td.__radd__(as6),  not timedelta.__sub__
+    # https://github.com/microsoft/pyright/issues/4088
     check(
-        assert_type(as6 + td, pd.Timedelta),  # pyright: ignore[reportGeneralTypeIssues]
+        assert_type(
+            as_dt_timedelta + td,  # pyright: ignore[reportGeneralTypeIssues]
+            pd.Timedelta,
+        ),
         pd.Timedelta,
     )
-    check(assert_type(as7 + td, pd.Timedelta), pd.Timedelta)
-    check(assert_type(as8 + td, pd.TimedeltaIndex), pd.TimedeltaIndex)
-    check(assert_type(as9 + td, pd.Series), pd.Series)
-    check(assert_type(as10 + td, pd.PeriodIndex), pd.PeriodIndex)
-    check(assert_type(as11 + td, pd.DatetimeIndex), pd.DatetimeIndex)
-    # pyright is wrong here because ndarray.__add__(Timedelta) is NotImplemented
+    check(assert_type(as_timedelta64 + td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(as_timedelta_index + td, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(as_timedelta_series + td, TimedeltaSeries), pd.Series)
+    check(assert_type(as_period_index + td, pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(as_datetime_index + td, pd.DatetimeIndex), pd.DatetimeIndex)
+    # pyright is fails here because ndarray.__add__(Timedelta) is NotImplemented
     check(
         assert_type(
-            as12 + td,  # pyright: ignore[reportGeneralTypeIssues]
+            as_ndarray_td64 + td,  # pyright: ignore[reportGeneralTypeIssues]
             npt.NDArray[np.timedelta64],
         ),
         np.ndarray,
+        np.timedelta64,
     )
-    check(assert_type(as14 + td, NaTType), NaTType)
+    check(assert_type(as_nat + td, NaTType), NaTType)
 
     # sub is not symmetric with dates. In general date_like - timedelta is
     # sensible, while timedelta - date_like is not
     # TypeError: as1, as2, as3, as4, as5, as10, as11, as13
```

Comment:
can you update the comment to refer to the newer names?

Also, you could do the `if TYPE_CHECKING_INVALID_USAGE` tests for these failures for completeness as you did for floor division.

Reply:
Fixed and added INVALID